server/src/main/scala/org/http4s/armeria/server/ArmeriaServerBuilder.scala

```diff
@@ -247,7 +253,12 @@ sealed class ArmeriaServerBuilder[F[_]] private (
     *   [[withTlsCustomizer]]
     */
   def withTls(keyCertChainFile: File, keyFile: File, keyPassword: Option[String]): Self =
-    atBuild(_.tls(keyCertChainFile, keyFile, keyPassword.orNull))
+    atBuild(
+      _.tlsProvider(
+        TlsProvider
+          .builder()
+          .keyPair(TlsKeyPair.of(keyFile, keyPassword.orNull, keyCertChainFile))
+          .build()))
```

Comment:
These changes addressing the armeria upgrade are quite trivial, but it'd be nice if @ikhoon took a quick look at it 🙇🏻‍♂️